### PR TITLE
[STACK-2262][STACK-2283] active-standby stabilize

### PR DESCRIPTION
**Pull request by ytsai-a10**
## Description
- Severity Level Critical
- Issue Description
For reload/reboot, vcs reload, vcs configuration sync, a10-octavia will sleep for 20 or 30 seconds. But a10-octavia should check both active and standby are up and vcs ready after the sleep or we will have problem for slow server.

## Jira Ticket
https://a10networks.atlassian.net/browse/STACK-2262
https://a10networks.atlassian.net/browse/STACK-2283

## Technical Approach
In active-standby mode, after reload/reboot, vcs reload or vcs configuration sync. check device is up and vcs ready after sleep.

## Config Changes
<pre>
<b>[vthunder]
default_vthunder_username = "admin"
default_vthunder_password = "a10"
default_axapi_version = "30"
#default_axapi_timeout = 600

[a10_controller_worker]
amp_image_owner_id = 99c9c2304f114685a32db30769c8a7e2
amp_secgroup_list = 2d0a3480-7f08-4184-b96f-c39bb444dd38
amp_flavor_id = 69995a83-c47a-427f-bf70-e9d9752aa915
amp_boot_network_list = 7f8ddd7f-c5c3-463a-b1c3-596f988480a4
amp_ssh_key_name = octavia_ssh_key
network_driver = a10_octavia_neutron_driver
workers = 2
amp_active_retries = 150
amp_active_wait_sec = 10
amp_busy_wait_sec = 0
amp_image_id = 0a845873-bd91-4f8c-ad58-cf5d5afde09c
#loadbalancer_topology = SINGLE
loadbalancer_topology = ACTIVE_STANDBY

[a10_health_manager]
udp_server_ip_address = 10.64.28.68
bind_port = 5550
bind_ip = 10.64.28.68
heartbeat_interval = 5
heartbeat_key = insecure
heartbeat_timeout = 90
health_check_interval = 3
failover_timeout = 600
health_check_timeout = 3
health_check_max_retries = 5

[a10_house_keeping]
load_balancer_expiry_age = 3600
amphorae_expiry_age = 3600
#use_periodic_write_memory = 'enable'
#write_mem_interval = 300

[a10_global]
network_type = "flat"
#network_type = "vlan"
vrid_floating_ip = "dhcp"
</b>
</pre>

## Test Cases
- create 2 LB with different subnet at the same time
- delete 2 LB with different subnet at the same time

Not tested yet: (TODO)
- create normal active-standby LB
- create members with different subnet
- delete members with different subnet

## Manual Testing
- Required: List of steps to manually test feature or fix

### CODE REVIEW

**Review by hthompson-a10 — DISMISSED** [review]

File: a10_octavia/controller/worker/tasks/vthunder_tasks.py [hunk 1]
```diff
@@ -1022,6 +1022,44 @@ def execute(self, vthunder):
...
 
+class VCSSyncWait(VThunderBaseTask):
+    """Task to wait VCS reload, VCS negotiagtion or VCS configuration sync ready."""
+
+    @axapi_client_decorator
+    def execute(self, vthunder):
+        attempts = CONF.a10_controller_worker.amp_vcs_retries
+        while attempts >= 0:
+            vmaster_ready = False
+            vblade_read = False
```
Comment (NehaKembalkarA10) [inline a10_octavia/controller/worker/tasks/vthunder_tasks.py:1033]: you can rename it as `vblade_ready`

Comment (ytsai-a10) [reply a10_octavia/controller/worker/tasks/vthunder_tasks.py]: fixed

File: a10_octavia/controller/worker/tasks/vthunder_tasks.py [hunk 2]
```diff
@@ -1022,6 +1028,45 @@ def execute(self, vthunder):
...
+                for i in range(len(vcs_member_list)):
+                    role = vcs_member_list[i]['state'].split('(')[0]
+                    if role == "vMaster":
+                        vmaster_ready = True
+                    if role == "vBlade":
+                        vblade_ready = True
+                    if vmaster_ready is True and vblade_ready is True:
+                        break
+                else:
+                    raise acos_errors.AxapiJsonFormatErrora(
```
Comment (hthompson-a10) [inline a10_octavia/controller/worker/tasks/vthunder_tasks.py:1054]: Typo. Should be `AxapiJsonFormatError`

Comment (ytsai-a10) [reply a10_octavia/controller/worker/tasks/vthunder_tasks.py:1054]: fixed in another PR

**Pull request merged**